### PR TITLE
Gk/194 db collection and queries for take action page

### LINKED ISSUE #194: DB Collection and Queries for Take Action page
## Description

The Take Action page will require a new collection, called Signatories, in our DB, and it will require 4 queries to interact with this collection.

The collection schema will look like
```Typescript
model Signatories {
  id           String        @id @default(auto()) @map("_id") @db.ObjectId
  firstName    String
  lastName     String
  title        Json[]
  institution  Json[]
  email        String        @unique
  createdAt    DateTime      @default(now())
  city         String?
  country      String?
  zipCode      Int?
  bioLink      String?
  twitter      String? 
  optIn        Boolean
}
```

and the queries will be

- `getSignatoriesCount`: return the number of signatories in the database
- `getSignatories`: return the First Name, Last Name, Titles, Institutions of each signatory, sorted by oldest createdAt.
- `addSignatory` : given inputs, insert into signatories database
- `deleteManySignatories` : given a signatory id, delete a signatory from the DB
- `updateUserEmailPreference`: given a signatory id and email preference, change the email pref in the DB

## Todo

- [x] Implement Schema
- [x] Implement `getSignatoriesCount`
- [x] Implement `getSignatories`
- [x] Implement `addSignatory`
- [x] Implement `deleteManySignatories` 
- [x] Implement `updateUserEmailPreference`

### PR DESCRIPTION
**Pull request by kennykos**
## Summary

Closes #194 

## Changes

- implements signatories schema
- implements queries into signatories DB

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -204,3 +204,19 @@ model WhitelistedUser {
   email  String  @unique
   userId String?
 }
+
+model Signatory {
+  id          String   @id @default(auto()) @map("_id") @db.ObjectId
+  firstName   String
+  lastName    String
+  title       Json[]
+  institution Json[]
```
Comment (zorazrr) [inline prisma/schema.prisma:213]: Maybe missing some context here but why are these two of `json[]` type?

**Review by zorazrr — COMMENTED** [review]
A small Q about the `Signatory` model. Otherwise it looks good. 
It'd also be good to add a piece of sample data to the DB and test some of the endpoints out on the frontend.

File: prisma/schema.prisma (hunk 1 above)
Comment (kennykos) [reply prisma/schema.prisma:213]: We want to allow people to have multiple titles/institutions, and prisma does not allow for a `String[]` type, so it seems like the best option for an array is `Json[]`

Comment (kennykos) [reply prisma/schema.prisma]: wow I don't know how I missed that, just changed in the latest commit. Also, I added an entry to the dev schema, is there a good way to text the endpoints?

**Comment by kennykos** [comment]
> Not related to this PR specifically but thought it'd be good to keep things consistent, we try to follow this contributing guideline:
> 
> > Start commit messages with a capital letter and use imperative mode (i.e. "Fix homepage" instead of "Fixes/Fixed homepage" - this is in line with GitHub best practices and auto-generated commit messages

Oh good to know, I will follow this from now on :vulcan_salute:

**Review by zorazrr — APPROVED** [review]

**Pull request merged**